nilearn/tests/test_extmath.py

```diff
@@ -6,13 +6,16 @@
 
 import numpy as np
 
+from sklearn.utils import check_random_state
 from nilearn._utils.extmath import fast_abs_percentile, is_spd
 
 
 def test_fast_abs_percentile():
-    data = np.arange(1, 100)
-    for p in range(10, 100, 10):
-        yield nose.tools.assert_equal, fast_abs_percentile(data, p - 1), p
+    rng = check_random_state(1)
+    data = np.arange(10, 100, 10)
+    rng.shuffle(data)
+    for p in data:
```

Comment:
In other words, I would still have `data` a shuffle of `np.arange(1, 100)` and `percentiles_to_check = range(10, 100, 10)`.